### PR TITLE
docs - pxf no longer requires hadoop client install

### PR DESCRIPTION
**Pull request by lisakowen**
doc updates related to removing PXF hadoop dependency:
- reorganized the subnav a bit, renamed a few sections, split some content to new pages
- no more hadoop client install required
- grouped all hadoop-related config (impersonation, kerberos) - into an optional subsection

question:
- if using hadoop connectors, instructing user to copy hadoop config files from hadoop cluster to /etc/xxx/conf directories on pxf segment hosts.  is this the desired location?  or $GPHOME/pxf/conf? or ??

doc staging site link:
- http://docs-lisa-pxfbundlejars.cfapps.io/600/pxf/instcfg_pxf.html  ("Configuring PXF" section)

### CODE REVIEW

File: gpdb-doc/markdown/pxf/client_instcfg.html.md.erb [hunk 1]
```diff
@@ -31,165 +24,61 @@ Perform the following procedure to install and configure the appropriate clients
...
         ```
         
-    2. Next, copy these Hadoop configuration files to each Greenplum Database segment host. For example:
+    2. Next, copy these Hadoop configuration files to the `/etc/hadoop/conf` directory on each Greenplum Database segment host. You may need to create this directory. For example:
 
         ``` shell
         gpadmin@gpmaster$ gpscp -v -f seghostfile core-site.xml =:/etc/hadoop/conf/core-site.xml
         gpadmin@gpmaster$ gpscp -v -f seghostfile hdfs-site.xml =:/etc/hadoop/conf/hdfs-site.xml
         gpadmin@gpmaster$ gpscp -v -f seghostfile mapred-site.xml =:/etc/hadoop/conf/mapred-site.xml
+        gpadmin@gpmaster$ gpscp -v -f seghostfile yarn-site.xml =:/etc/hadoop/conf/yarn-site.xml
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/client_instcfg.html.md.erb]: Should we use $GPHOME/pxf/conf/ as the suggested path to copy hadoop XMLs? (i.e.)
```suggestion
        gpadmin@gpmaster$ gpscp -v -f seghostfile yarn-site.xml =:$GPHOME/pxf/conf/yarn-site.xml
```

File: gpdb-doc/markdown/pxf/client_instcfg.html.md.erb [hunk 2]
```diff
@@ -31,165 +24,61 @@ Perform the following procedure to install and configure the appropriate clients
...
+2. If you plan to use the PXF Hive connector to access Hive table data, copy Hive configuration to each Greenplum Database segment host.
     
     1. Copy the `hive-site.xml` Hive configuration file from one of the hosts on which your Hive service is running to the current host. For example:
 
         ``` shell
         gpadmin@gpmaster$ scp hiveuser@hivehost:/etc/hive/conf/hive-site.xml .
         ```
 
-    2. Next, copy the `hive-site.xml` configuration file to each Greenplum Database segment host. For example:
+    2. Next, copy the `hive-site.xml` configuration file to the `/etc/hive/conf` directory on each Greenplum Database segment host. You may need to create this directory. For example:
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/client_instcfg.html.md.erb]: Again, I think we should use the conf directory inside $GPHOME/pxf/
```suggestion
    2. Next, copy the `hive-site.xml` configuration file to the `$GPHOME/pxf/conf` directory on each Greenplum Database segment host. You may need to create this directory. For example:
```

File: gpdb-doc/markdown/pxf/client_instcfg.html.md.erb [hunk 3]
```diff
@@ -31,165 +24,61 @@ Perform the following procedure to install and configure the appropriate clients
...
+3. If you plan to use the PXF HBase connector to access HBase table data, copy HBase configuration to each Greenplum Database segment host.
     
     1. Copy the `hbase-site.xml` HBase configuration file from one of the hosts on which your HBase service is running to the current host. For example:
 
         ``` shell
         gpadmin@gpmaster$ scp hbaseuser@hbasehost:/etc/hive/conf/hbase-site.xml .
         ```
 
-    2. Next, copy the `hbase-site.xml` configuration file to each Greenplum Database segment host. For example:
+    2. Next, copy the `hbase-site.xml` configuration file to the `/etc/hbase/conf` directory on each Greenplum Database segment host. You may need to create this directory. For example:
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/client_instcfg.html.md.erb]:
```suggestion
    2. Next, copy the `hbase-site.xml` configuration file to the `$GPHOME/pxf/conf` directory on each Greenplum Database segment host. You may need to create this directory. For example:
```

File: gpdb-doc/markdown/pxf/install_java.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    1. Install the Java package. For example, to install Java version 1.8:
+
+        ``` shell
+        gpadmin@gpmaster$ gpssh -e -v -f seghostfile sudo yum -y install java-1.8.0-openjdk-1.8.0*
+        ```
+
+    2. Identify the Java base install directory. Update the `gpadmin` user's `.bash_profile` file on each segment host to include this `$JAVA_HOME` setting if it is not already present. For example:
+
+        ``` shell
+        gpadmin@gpmaster$ gpssh -e -v -f seghostfile "echo 'export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.144-0.b01.el7_4.x86_64/jre' >> /home/gpadmin/.bash_profile"
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/install_java.html.md.erb]: We shouldn't suggest using the minor java version as these versions change frequently

```suggestion
        gpadmin@gpmaster$ gpssh -e -v -f seghostfile "echo 'export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk.x86_64/jre' >> /home/gpadmin/.bash_profile"
```

File: gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb [hunk 5]
```diff
@@ -73,7 +73,7 @@ After you upgrade to the new version of Greenplum Database, perform the followin
 
 4. If you updated any of the `pxf-profiles.xml`, `pxf-log4j.properties`, `pxf-private.classpath`, or `pxf-public.classpath` configuration files in your *PXF.from* installation, re-apply those changes and copy the updated file(s) to all segment hosts. Refer to Step 3 above for a similar `gpscp` command.
 
-    **Note:** Starting in Greenplum Database version 5.12, the package name for PXF classes was changed to use the prefix `org.greenplum.*`. If you are upgrading from an older *PXF.from* version and you customized the `pxf-profiles.xml` file, you must change any `org.apache.hawq.*` references to `org.greenplum.*` when you re-apply your changes.
+    **Note:** Starting in Greenplum Database version 5.12.0, the package name for PXF classes was changed to use the prefix `org.greenplum.*`. If you are upgrading from an older *PXF.from* version and you customized the `pxf-profiles.xml` file, you must change any `org.apache.hawq.*` references to `org.greenplum.*` when you re-apply your changes.
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/upgrade_pxf.html.md.erb]: I think this should be as follows:
```suggestion
    **Note:** Starting in Greenplum Database version 5.12.0, the package name for PXF classes was changed to use the prefix `org.greenplum.pxf.*`. If you are upgrading from an older *PXF.from* version and you customized the `pxf-profiles.xml` file, you must change any `org.apache.hawq.pxf.*` references to `org.greenplum.pxf.*` when you re-apply your changes.
```

**Comment by lisakowen** [comment]
@frankgh @lavjain @kongyew @denalex - i updated the hadoop config page.  see: http://docs-lisa-pxfbundlejars.cfapps.io/600/pxf/client_instcfg.html.  the page still instructs the user to copy the config files from their hadoop cluster.

re: having the user edit the $GPHOME/pxf/conf/xxx-site.xml template files directly - wouldn't the users obtain the values for the properties they need to set in those files from their hadoop configuration anyway?  i think copying the files would be simpler.  but i might be missing something here.

**Pull request merged**